starfish/imagestack/parser/crop.py

```diff
@@ -43,6 +43,14 @@ def __init__(
         self._x_slice = x_slice
         self._y_slice = y_slice
 
+    def add_permitted_axes(self, axis_type: Axes, permitted_axis: int) -> None:
```

Comment:
Am I misunderstanding this function? It seems like this would silently fail if any axis such as `Axes.CH` evaluates to `None`.

I wonder if this could be simplified by adding an assert statement somewhere upstream in the code about an inconsistent state, similar to what Tony's done elsewhere, and make this fail explicitly.

Reply:
You actually have to do the check here of the linter gets mad with `error: Item "None" of "Optional[Set[int]]" has no attribute "add"`

Reply:
I am kind of on the fence whether this method should flag an error if the `_permitted_{axis}` is `None` to begin with.  That almost suggests user error.

I would also consider making `add_permitted_axes` a "private" method (a.k.a., `_add_permitted_axes`).